src/RelayClient.ts

```diff
@@ -633,6 +643,39 @@ export class RelayClient {
                 relayInfo
             )} transaction: ${JSON.stringify(transactionDetails)}`
         );
+        if ([undefined, null].includes(transactionDetails.tokenAmount)) {
+            log.debug('Calculating maxPossibleGas...');
+
+            const txDetailsClone: EnvelopingTransactionDetails = {
+                ...transactionDetails,
+                // a dump value, just to allow the correct estimation, including the token transfer
+                tokenAmount: web3.utils.toWei('1').toString()
```

Comment:
a couple of questions:
1. does this fail without a `tokenAmount` set?
2. does it make a difference which value we use? would `0` work?

Reply:
No, it doesn't fail, but it's required in order to estimate the complete workflow. Without the `tokenAmount` set, it will estimate the flow where the `tokenAmount` is not set. For instance, have a look at [the SmartWallet](https://github.com/anarancio/rif-relay-contracts/blob/master/contracts/smartwallet/SmartWallet.sol#L113) contract

Reply:
got it. makes perfect sense. thanks!